pumgen crashes if properties given by a LayeredModel (VelocityAwareMeshing)

I've noticed that VelocityAwareMeshing seems to always crash with the velocity is given by a LayeredModel.
Here is an extract of the log:
```
Wed Feb 22 14:42:28, Info:  face id: 116 , MSize = 1000 
Wed Feb 22 14:42:28, Info:  face id: 120 , MSize = 1000 
Wed Feb 22 14:42:28, Info:  face id: 122 , MSize = 1000 
Wed Feb 22 14:42:28, Info:  Enabling velocity aware meshing 
Wed Feb 22 14:42:28, Info:  Gradation rate = 0.8 
Wed Feb 22 14:42:28, Info:  Target equivolume AspectRatio = 12 
Wed Feb 22 14:42:28, Info:  Target equiarea AspectRatio = 6 
Wed Feb 22 14:42:28, Info:  Starting the surface mesher 
Wed Feb 22 14:42:28, Info:  Progress: Surface Meshing 
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun noticed that process rank 8 with PID 0 on node exception exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
```
Here is the xml file:
```
<freeSurface>119</freeSurface>
<!-- hypocenter-->
<boundaryCondition tag="65">121</boundaryCondition>
<!-- fault 7.8-->
<boundaryCondition tag="66">101,107,112,114,121</boundaryCondition>
<!-- fault 7.5-->
<boundaryCondition tag="67">97,109,115,116,120,122</boundaryCondition>
<!-- extras fault 7.8 -->
<boundaryCondition tag="68">110,105</boundaryCondition>
<absorbing>93,95,103,104,108</absorbing>
<surfaceMSize value="1000">101,107,112,114,121,110,105</surfaceMSize>
<surfaceMSize value="1000">97,109,115,116,120,122</surfaceMSize>
<area_AspectRatio value="6"/>
<vol_AspectRatio value="12"/>
<SurfaceMeshing SmoothingLevel="1" SmoothingType="Gradient" DiscreteAngle="15.0" Snap="0"></SurfaceMeshing>
<VolumeMeshing  SmoothingLevel="1" SmoothingType="Gradient" SetOptimisation="1"></VolumeMeshing>
<gradation value="0.8"/>
<globalMSize value="20e3"/>
<VelocityAwareMeshing easiFile="Mw_78_Turkey_rhomulambda1D_Guvercin_et_al.yaml" elementsPerWaveLength="3">
    <VelocityRefinementCuboid frequency="0.1" centerX="0" centerY="0" centerZ="-15e3"
                              halfSizeX="40e10" halfSizeY="15e10" halfSizeZ="15e10" bypassFindRegionAndUseGroup="1"/>
</VelocityAwareMeshing>
```
and here is the yaml file:
```
!LayeredModel
  map: !AffineMap
    matrix:
      z1: [0.0, 0.0, 1.0]
    translation:
      z1: 0
  interpolation: lower
  parameters: [rho, mu, lambda, plastCo]
  nodes:
      -1000.0: [2464.9689710139387, 14555395276.94021, 21249511511.522747, 5822158.1107760845]
      -1000000.0: [3400.203751580001, 72261470148.95335, 89717096148.43964, 28904588.059581343]
```
if I use a LuaMap instead:
```
!LuaMap
returns: [rho, mu, lambda]
function: |
 function f (x)
    if (x["z"] >= -1000.0) then
      rho = 2464.9689710139387
      mu = 14555395276.94021
      lambda = 21249511511.522747
    elseif (x["z"] >= -1000000.0) then
      rho = 3400.203751580001
      mu = 72261470148.95335
      lambda = 89717096148.43964
    end
   return {
     rho = rho,
     mu = mu,
     lambda = lambda
   }
   end
```

I'm able to generate the mesh.

the setup is available on exception, at /export/dump/ulrich/PUMGEN_folder/bug_layered_model

fix-53 does not compile
```
     86     /tmp/ulrich/spack-stage/spack-stage-pumgen-fix-53-pdiucxoyuqhncyllhvnzrg5dh4kxa3yy/spack-src/src/input/EasiMeshSize.cpp: In member function 'int EasiMeshSize::findGroup(const std::array<double, 3>&)':
  >> 87     /tmp/ulrich/spack-stage/spack-stage-pumgen-fix-53-pdiucxoyuqhncyllhvnzrg5dh4kxa3yy/spack-src/src/input/EasiMeshSize.cpp:26:44: error: invalid conversion from 'std::array<double, 3>::const_pointer' {aka 'const double*'} to 'double*' [-fperm
            issive]
     88        26 |     if (GR_containsPoint(region, point.data()) > 0) {
     89           |                                  ~~~~~~~~~~^~
     90           |                                            |
     91           |                                            std::array<double, 3>::const_pointer {aka const double*}
     92     In file included from /tmp/ulrich/spack-stage/spack-stage-pumgen-fix-53-pdiucxoyuqhncyllhvnzrg5dh4kxa3yy/spack-src/src/input/EasiMeshSize.h:6,
     93                      from /tmp/ulrich/spack-stage/spack-stage-pumgen-fix-53-pdiucxoyuqhncyllhvnzrg5dh4kxa3yy/spack-src/src/input/EasiMeshSize.cpp:1:
     94     /import/heisenbug-dump/ulrich/myLibs/SimLib/2024.0-240616/include/SimModel.h:393:60: note:   initializing argument 2 of 'int GR_containsPoint(pGRegion, double*)'
     95       393 | Model_EXPORT int GR_containsPoint(pGRegion region, double *pnt);
     96           |                                                    ~~~~~~~~^~~
  >> 97     make[2]: *** [CMakeFiles/pumgen.dir/build.make:219: CMakeFiles/pumgen.dir/src/input/EasiMeshSize.cpp.o] Error 1
     98     make[2]: *** Waiting for unfinished jobs....
     99     make[2]: Leaving directory '/tmp/ulrich/spack-stage/spack-stage-pumgen-fix-53-pdiucxoyuqhncyllhvnzrg5dh4kxa3yy/spack-build-pdiucxo'
  >> 100    make[1]: *** [CMakeFiles/Makefile2:88: CMakeFiles/pumgen.dir/all] Error 2
     101    make[1]: Leaving directory '/tmp/ulrich/spack-stage/spack-stage-pumgen-fix-53-pdiucxoyuqhncyllhvnzrg5dh4kxa3yy/spack-build-pdiucxo'
  >> 102    make: *** [Makefile:139: all] Error 2
```

--- a/.github/workflows/cmake.yml
+++ b/.github/workflows/cmake.yml
@@ -17,7 +17,7 @@ jobs:
       - name: Install dependencies
         run: |
           sudo apt-get update
-          sudo apt-get install -qq g++ openmpi-bin openmpi-common libopenmpi-dev hdf5-tools libhdf5-openmpi-103 libhdf5-openmpi-dev libnetcdf-dev
+          sudo apt-get install -qq g++ openmpi-bin openmpi-common libopenmpi-dev hdf5-tools libhdf5-openmpi-dev libnetcdf-dev
   
       - name: Configure PUMGen
         working-directory: ${{ github.workspace }}

--- a/src/input/EasiMeshSize.cpp
+++ b/src/input/EasiMeshSize.cpp
@@ -14,7 +14,7 @@ EasiMeshSize::EasiMeshSize(VelocityAwareRefinementSettings refinementSettings, p
   model = parser->parse(refinementSettings.getEasiFileName());
 }
 
-int EasiMeshSize::findGroup(const std::array<double, 3>& point) {
+int EasiMeshSize::findGroup(std::array<double, 3> point) {
   // GR_containsPoint can be expensive for large geometry,
   // therefore we bypass it the simple case of one region
   if (groupMap.size() == 1) {

--- a/src/input/EasiMeshSize.cpp
+++ b/src/input/EasiMeshSize.cpp
@@ -1,19 +1,20 @@
 #include "EasiMeshSize.h"
+#include <unordered_map>
 #include <utils/logger.h>
 
 #include <tuple>
 #include <utility>
 
-EasiMeshSize::EasiMeshSize() : parser(nullptr), model(nullptr), query(easi::Query(1, 3)) {}
+EasiMeshSize::EasiMeshSize() : parser(nullptr), model(nullptr) {}
 
 EasiMeshSize::EasiMeshSize(VelocityAwareRefinementSettings refinementSettings, pGModel simModel,
                            std::unordered_map<pGRegion, int> groupMap)
     : refinementSettings(refinementSettings), simModel(simModel), parser(new easi::YAMLParser(3)),
-      groupMap(std::move(groupMap)), query(easi::Query(1, 3)) {
+      groupMap(std::move(groupMap)) {
   model = parser->parse(refinementSettings.getEasiFileName());
 }
 
-int EasiMeshSize::findGroup(std::array<double, 3> point) {
+int EasiMeshSize::findGroup(const std::array<double, 3>& point) {
   // GR_containsPoint can be expensive for large geometry,
   // therefore we bypass it the simple case of one region
   if (groupMap.size() == 1) {
@@ -31,7 +32,7 @@ int EasiMeshSize::findGroup(std::array<double, 3> point) {
 }
 
 std::tuple<const double, const int>
-EasiMeshSize::getTargetedFrequencyAndRegion(std::array<double, 3> point) {
+EasiMeshSize::getTargetedFrequencyAndRegion(const std::array<double, 3>& point) {
   const auto& refinementRegions = refinementSettings.getRefinementRegions();
   double targetedFrequency = 0.0;
   int bypassFindRegionAndUseGroup = 0;
@@ -58,7 +59,7 @@ EasiMeshSize::getTargetedFrequencyAndRegion(std::array<double, 3> point) {
   return std::make_tuple(targetedFrequency, bypassFindRegionAndUseGroup);
 }
 
-double EasiMeshSize::getMeshSize(std::array<double, 3> point) {
+double EasiMeshSize::getMeshSize(const std::array<double, 3>& point) {
   if (!model) {
     logError() << "Model was not parsed correctly";
   }
@@ -71,6 +72,7 @@ double EasiMeshSize::getMeshSize(std::array<double, 3> point) {
     return defaultMeshSize;
   }
 
+  easi::Query query(1, 3);
   query.x(0, 0) = point[0];
   query.x(0, 1) = point[1];
   query.x(0, 2) = point[2];

--- a/src/input/EasiMeshSize.h
+++ b/src/input/EasiMeshSize.h
@@ -20,13 +20,13 @@ class EasiMeshSize {
 
   easi::YAMLParser* parser;
   easi::Component* model; // Unique ptr to model leads to segfault
-  easi::Query query;
   pGModel simModel;
   std::unordered_map<pGRegion, int> groupMap;
 
-  int findGroup(std::array<double, 3> point);
+  int findGroup(const std::array<double, 3>& point);
 
-  std::tuple<const double, const int> getTargetedFrequencyAndRegion(std::array<double, 3> point);
+  std::tuple<const double, const int>
+  getTargetedFrequencyAndRegion(const std::array<double, 3>& point);
 
   public:
   EasiMeshSize();
@@ -34,7 +34,7 @@ class EasiMeshSize {
   EasiMeshSize(VelocityAwareRefinementSettings refinementSettings, pGModel simModel,
                std::unordered_map<pGRegion, int> groupMap);
 
-  double getMeshSize(std::array<double, 3> point);
+  double getMeshSize(const std::array<double, 3>& point);
 };
 
 #endif // PUMGEN_EASIMESHSIZE_H

--- a/src/input/EasiMeshSize.h
+++ b/src/input/EasiMeshSize.h
@@ -23,7 +23,7 @@ class EasiMeshSize {
   pGModel simModel;
   std::unordered_map<pGRegion, int> groupMap;
 
-  int findGroup(const std::array<double, 3>& point);
+  int findGroup(std::array<double, 3> point);
 
   std::tuple<const double, const int>
   getTargetedFrequencyAndRegion(const std::array<double, 3>& point);

--- a/src/input/SimModSuite.h
+++ b/src/input/SimModSuite.h
@@ -69,7 +69,7 @@
  */
 class SimModSuite : public FullStorageMeshData {
   private:
-  EasiMeshSize easiMeshSize;
+  std::shared_ptr<EasiMeshSize> easiMeshSize;
   pGModel m_model;
 
   pParMesh m_simMesh;
@@ -587,7 +587,8 @@ class SimModSuite : public FullStorageMeshData {
 
     if (MeshAtt.velocityAwareRefinementSettings.isVelocityAwareRefinementOn()) {
       logInfo(PMU_rank()) << "Enabling velocity aware meshing";
-      easiMeshSize = EasiMeshSize(MeshAtt.velocityAwareRefinementSettings, model, groupMap);
+      easiMeshSize =
+          std::make_shared<EasiMeshSize>(MeshAtt.velocityAwareRefinementSettings, model, groupMap);
       auto easiMeshSizeFunc = [](pSizeAttData sadata, void* userdata) {
         auto* easiMeshSize = static_cast<EasiMeshSize*>(userdata);
         std::array<double, 3> pt{};
@@ -598,7 +599,7 @@ class SimModSuite : public FullStorageMeshData {
         return easiMeshSize->getMeshSize(pt);
       };
       // set the user-defined function for isotropic size
-      MS_setSizeAttFunc(meshCase, "setCustomMeshSize", easiMeshSizeFunc, &easiMeshSize);
+      MS_setSizeAttFunc(meshCase, "setCustomMeshSize", easiMeshSizeFunc, easiMeshSize.get());
       // Relative anisotropic size is set for the entire model through the
       // function anisoSize
       MS_setMeshSize(meshCase, GM_domain(model), MS_userDefinedType | 1, 0, "setCustomMeshSize");

